### PR TITLE
texture cube

### PR DESCRIPTION
**Pull request by shrekshao**
- Mouse Input
- Render Loop
- helper function: load multi images
- use gl-Matrix
- a skybox demo, not a exact port from ogl sample, because: 
  - TEXTURE_CUBE_MAP_SEAMLESS is not found in webgl2 document
  - I did not fully understand the meaning of ogl sample, for it using a constant normal vector
  - I think a skybox demo is more direct to demonstrate cubemap

### CODE REVIEW

**Comment by shrekshao** [comment]
Since I use gl-Matrix min, which fail the travis test, is there a way to ignore this file? @pjcozzi

File: README.md [hunk 1]
```diff
@@ -78,3 +79,4 @@ When you open a pull request, please make sure that GitHub reports that "All che
 * [OpenGL Samples Pack](https://github.com/g-truc/ogl-samples) by Christophe Riccio ([@Groovounet](https://github.com/Groovounet)) and contributors
 * [Cesium](https://github.com/AnalyticalGraphicsInc/cesium) build script by Matt Amato ([@mramato](https://github.com/mramato)) and contributors
 * [webgl2-particles](https://github.com/toji/webgl2-particles) by Brandon Jones ([@toji](https://github.com/toji)) and Mr.doob ([@mrdoob](https://github.com/mrdoob))
+* [gl-Matrix](https://github.com/toji/gl-matrix) by Brandon Jones ([@toji](https://github.com/toji)) and contributors
```
Comment (pjcozzi) [inline README.md:83]: Move this to LICENSE.md, and include it in the third-party section like the others.

**Comment by pjcozzi** [comment]
> Since I use gl-Matrix min, which fail the travis test, is there a way to ignore this file

If it's easy, fix the warnings and open a pull request for that project.

Otherwise, ignore it by modifying this line of the build script: https://github.com/WebGLSamples/WebGL2Samples/blob/master/gulpfile.js#L7

**Comment by pjcozzi** [comment]
> samples/gl-matrix-min.js

Move this to a sub-directory, e.g., `samples/third-party/gl-matrix-min.js`

File: samples/texture_cube.html [hunk 2]
```diff
@@ -0,0 +1,237 @@
+<!DOCTYPE html>
+<!-- 
+    Reference: OpenGL Samples Pack https://github.com/g-truc/ogl-samples/blob/master/tests/gl-320-glsl-discard.cpp
```
Comment (pjcozzi) [inline samples/texture_cube.html]: Is this the right .cpp file?

**Comment by pjcozzi** [comment]
What is specific to WebGL 2 about this sample?

I don't think `TEXTURE_CUBE_MAP_SEAMLESS` is in ES 3.0 so it will not be in WebGL 2.

See Sections 3.8.9 and F.2 of the ES 3.0 spec (in general, please check the ES spec when in doubt).

> OpenGL ES 3.0 requires that all cube map filtering be seamless. OpenGL ES
> 2.0 specified that a single cube map face be selected and used for filtering

**Comment by shrekshao** [comment]
Yeah, I agree with `TEXTURE_CUBE_MAP_SEAMLESS` is not available in WebGL 2. In light of this, merely using `TEXTURE_CUBE_MAP` is meaningless for WebGL 2. Shall I close this pull request later?

Anyway, with this try we now have : 
- Mouse Control
- Render loop
- gl-Matrix support

Not bad

**Comment by shrekshao** [comment]
For Seamless cube map, https://docs.google.com/presentation/d/1Orx0GB0cQcYhHkYsaEcoo5js3c5-pv7ahPniIRIzzfg/edit#slide=id.gd15060520_0_38
It is mentioned in WebGL 2 spec, 
yet I can't find it in either ES 3 or WebGL 2 conformance test

**Comment by pjcozzi** [comment]
@shrekshao the enum is not in the spec because it is always enabled.  As I said above:

See Sections 3.8.9 and F.2 of the ES 3.0 spec (in general, please check the ES spec when in doubt).

> OpenGL ES 3.0 requires that all cube map filtering be seamless. OpenGL ES
> 2.0 specified that a single cube map face be selected and used for filtering

**Comment by shrekshao** [comment]
Oh oh, I misunderstand it. Besides this, I fail to find anything specific to WebGL 2

**Comment by pjcozzi** [comment]
OK, let's delete the sample and keep any potential useful code in this branch until we need it - then we can `git cherry-pick` the commits.

**Comment by kenrussell** [comment]
FYI: es3fShaderTextureFunctionTests.js , es3fTextureFilteringTests.js and es3fTextureShadowTests.js should be testing whether cube maps are seamless in WebGL 2.0.

**Comment by shrekshao** [comment]
Hi @kenrussell , I took a look at these js files. However I noticed that the "seamless flag" are all related to the testing framework, but I failed to find any GL calls that setting the TEXTURE_CUBE_MAP_SEAMLESS. So I am a bit confused here. I don't understand why there's no such a call.

**Comment by kenrussell** [comment]
The reason there's no such parameter is that cube maps are always seamless in OpenGL ES 3.0 and there's no way to disable that behavior.

**Comment by shrekshao** [comment]
Ok.. Thank you for the clarification!

**Pull request closed without merging**